aten/src/ATen/native/native_functions.yaml

```diff
@@ -6625,6 +6625,24 @@
     CPU: legacy::cpu::_th_histc
     CUDA: _histc_cuda
 
+- func: histogram.bins_tensor_out(Tensor self, Tensor bins, Scalar min=0, Scalar max=0, Tensor? weight=None, bool density=False, *, Tensor(a!) hist, Tensor(b!) bin_edges) -> (Tensor(a!) hist, Tensor(b!) bin_edges)
+  dispatch:
+    CPU: histogram_out_cpu
+
+- func: histogram.bins_tensor(Tensor self, Tensor bins, Scalar min=0, Scalar max=0, Tensor? weight=None, bool density=False) -> (Tensor hist, Tensor bin_edges)
+  variants: method, function
+  dispatch:
+    CPU: histogram_cpu
+
+- func: histogram.bin_ct_out(Tensor self, int bins=100, Scalar min=0, Scalar max=0, Tensor? weight=None, bool density=False, *, Tensor(a!) hist, Tensor(b!) bin_edges) -> (Tensor(a!) hist, Tensor(b!) bin_edges)
+  dispatch:
+    CPU: histogram_out_cpu
+
+- func: histogram.bin_ct(Tensor self, int bins=100, Scalar min=0, Scalar max=0, Tensor? weight=None, bool density=False) -> (Tensor hist, Tensor bin_edges)
```

Comment:
NumPy and PyTorch were developed with support for Python 2, but now PyTorch only supports Python 3+. This means every version of Python we support allows for keyword-only arguments. The newer Python Array API aggressively separates arguments into positional-only and keyword-only (see https://data-apis.org/array-api/latest/). Is there a natural place to make some of these arguments keyword-only?

Note that PyTorch doesn't yet support positional-only arguments because those aren't available until Python 3.8, and PyTorch still supports earlier Python versions.